lib/loaders/images.js

```diff
@@ -0,0 +1,63 @@
+/*
+ * This file is part of the Symfony Webpack Encore package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+'use strict';
+
+const WebpackConfig = require('../WebpackConfig'); //eslint-disable-line no-unused-vars
+const loaderFeatures = require('../features');
+const applyOptionsCallback = require('../utils/apply-options-callback');
+
+module.exports = {
+    /**
+     * @param {WebpackConfig} webpackConfig
+     * @return {Array} of loaders to use for images
+     */
+    getLoaders(webpackConfig) {
+
+        // Default filename can be overridden using Encore.configureFilenames({ images: '...' })
+        let filename = 'images/[name].[hash:8].[ext]';
+        if (webpackConfig.configuredFilenames.images) {
+            filename = webpackConfig.configuredFilenames.images;
+        }
+
+        // The url-loader can be used instead of the default file-loader by
+        // calling Encore.configureUrlLoader({ images: {/* ... */}})
+        let loaderName = 'file-loader';
+        const loaderOptions = {
+            name: filename,
+            publicPath: webpackConfig.getRealPublicPath()
+        };
+
+        if (webpackConfig.urlLoaderOptions.images) {
+            loaderFeatures.ensurePackagesExistAndAreCorrectVersion('urlloader');
+            loaderName = 'url-loader';
+            Object.assign(loaderOptions, webpackConfig.urlLoaderOptions.images);
+        }
+
+        const imagesLoaders = [{
+            loader: loaderName,
+            options: loaderOptions
+        }];
+
+        if (webpackConfig.useImagemin) {
+            loaderFeatures.ensurePackagesExistAndAreCorrectVersion('imagemin');
+
+            const imageminOptions = {
+                disable: !webpackConfig.isProduction(),
```

Comment:
Hmm, I'm not sure about this. I could see a user enabling this feature while developing, then checking the images and saying "yep! they look ok still!. Then on deploy, their images are now a "lower quality". I think we should just allow the user to control whether or not this is disabled entirely.

Reply:
🤔  that should just be the default value, and one can just override that in the webpack/encore configuration?

Reply:
If I call `Encore.enableImagemin()`, I would expect that my images would be optimized. NOT optimizing them is unexpected. So I think Encore needs to not have this `disable` option at all. Nice and simple: if they call `Encore.enableImagemin()`, we enable it. Then, if they choose to, they could decide to disable in dev.

I *do* realize why you're doing this... it's just too unexpected. One other option would be to add an `isEnabled` flag as the *first* argument to `Encore.enableImagemin()`. Then we would document you using it like this:

```
Encore.enableImagemin(Encore.isProduction());
```

Reply:
Sorry for the delay. I decided to remove the disabled-in-dev default. You're probably right that it's unexpected, and it's easy to add. I thought about the additional option as well, but it just adds unnecessary complexity, I can achieve the same using an options object.